app/uk/gov/hmrc/rasapi/services/ResultsGenerator.scala

```diff
@@ -38,27 +38,47 @@ trait ResultsGenerator {
   def getCurrentDate: DateTime
   val allowDefaultRUK: Boolean
 
+  val TOO_MANY_REQUESTS = "TOO_MANY_REQUESTS"
+  val DECEASED = "DECEASED"
+  val MATCHING_FAILED = "MATCHING_FAILED"
+  val INTERNAL_SERVER_ERROR = "INTERNAL_SERVER_ERROR"
+  val REQUEST_TIMED_OUT = "REQUEST_TIMEOUT"
+  val BAD_REQUEST = "BAD_REQUEST"
+
+  val retryLimit = 3
+  val _1Second = 1000L
+
   def fetchResult(inputRow:String, userId: String)(implicit hc: HeaderCarrier, request: Request[AnyContent]):String = {
+
+    def getResultAndProcess(memberDetails: IndividualDetails, retryCount:Int = 1): String = {
+
+      val res = Await.result(desConnector.getResidencyStatus(memberDetails, userId),20 second)
+
+      res match {
+        case Left(residencyStatus) =>
+          val resStatus = if (residencyYearResolver.isBetweenJanAndApril()) updateResidencyResponse(residencyStatus)
+          else residencyStatus.copy(nextYearForecastResidencyStatus = None)
+          auditResponse(failureReason = None, nino = Some(memberDetails.nino),
+            residencyStatus = Some(resStatus), userId = userId)
+          inputRow + comma + resStatus.toString
+
+        case Right(statusFailure) =>
+          if (statusFailure.code != DECEASED && statusFailure.code != MATCHING_FAILED && retryCount <= retryLimit) {
+            Thread.sleep(_1Second)
+            getResultAndProcess(memberDetails, retryCount = retryCount + 1)
+          } else {
+            auditResponse(failureReason = Some(statusFailure.code), nino = Some(memberDetails.nino),
+              residencyStatus = None, userId = userId)
+            inputRow + comma + statusFailure.code.replace(DECEASED, MATCHING_FAILED).replace(TOO_MANY_REQUESTS, INTERNAL_SERVER_ERROR)
```

Comment:
might have to replace error_RequestTimeout as well to Internal server error, I suggest only use INTERNAL SERVER ERROR so dont have to replace